RFS/src/main/java/org/opensearch/migrations/reindexer/FailedRequestsLogger.java

```diff
@@ -0,0 +1,58 @@
+package org.opensearch.migrations.reindexer;
+
+import java.util.Optional;
+import java.util.function.IntSupplier;
+import java.util.function.Supplier;
+
+import com.rfs.common.OpenSearchClient.OperationFailed;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class FailedRequestsLogger {
+    static final String LOGGER_NAME = "FailedRequestsLogger";
+    private final Logger logger = LoggerFactory.getLogger(LOGGER_NAME);
+
+    public void logBulkFailure(
+        String indexName,
+        IntSupplier failedItemCounter,
+        Supplier<String> bulkRequestBodySupplier,
+        Throwable error
+    ) {
+        var rootCause = getRootCause(error);
+
+        var responseBody = Optional.ofNullable(rootCause)
+            .filter(OperationFailed.class::isInstance)
+            .map(OperationFailed.class::cast)
+            .map(opFailed -> opFailed.response)
+            .map(response -> response.body);
+
+        if (responseBody.isPresent()) {
+            logger.atInfo()
+                .setMessage(
+                    "Bulk request failed for {} index on {} items, reason {}, bulk request body followed by response:\n{}\n{}"
```

Comment:
nit: Do we have the shard here? Would be helpful to know which shard to retry

Reply:
Nope, it seems strange to pipe that through as is - we can revisit this as we iterate on logging